Sources/NIO/BaseSocket.swift

```diff
@@ -218,7 +214,7 @@ class BaseSocket: BaseSocketProtocol {
     private var descriptor: NIOBSDSocket.Handle
     public var isOpen: Bool {
         #if os(Windows)
-            return descriptor != WinSDK.INVALID_SOCKET
+            return descriptor != NIOBSDSocket.invalidHandle
```

Comment:
Can we blast away the whole `#if-#else-#endif` block here leaving just this line? Or do we expect negative descriptors which aren't `-1`?

Reply:
I think we're just being extra cautious. We could get rid of it, but I'd prefer to do it by factoring out this code entirely but keeping the defensiveness.

Reply:
Sounds reasonable.